hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java

```diff
@@ -3425,6 +3438,12 @@ private void scan(HBaseRpcController controller, ScanRequest request, RegionScan
             }
             boolean mayHaveMoreCellsInRow = scannerContext.mayHaveMoreCellsInRow();
             Result r = Result.create(values, null, stale, mayHaveMoreCellsInRow);
+
+            if (request.getScan().getResultMetricsEnabled()) {
```

Comment:
What's the reason for not directly setting it on the result for scans?

Reply:
good question, scans are a bit tricky just due to how the results are constructed on the client side. you can take a look at how the [response converter](https://github.com/apache/hbase/pull/6623/files#diff-8ff91ab247b0d6cd0e97bc279ede64b55db8f31512ef2405eb5f68485b395fd6R453) builds these responses. They're created from the cells, so we need to send the metrics separately through the wire in this case and hydrate them client-side.